### PR TITLE
postfix 3.9

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: postfix-3.9.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        local infile="$3"
+
+        # Use sed to find the line matching the regex, write it to outfile, and delete it from infile
+        sed -i -E -e "\|$regex|{
+      w $outfile
+      d
+      }" "$infile"
+      }
+
+      addgroup -S postfix 2>/dev/null
```
Comment (imjasonh) [inline postfix-3.9.yaml:62]: Will the image need this user+group too?

File: postfix-3.9.yaml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+data:
+  - name: _subpackages
+    items:
+      ldap:
+      mysql:
+      pcre:
+      pgsql:
+      sqlite:
+
+pipeline:
```
Comment (imjasonh) [inline postfix-3.9.yaml:42]: Could you add a test too?

File: postfix-3.9.yaml (hunk 1 above)
Comment (developer-guy) [reply postfix-3.9.yaml:62]: this is package only type of request afaik

File: postfix-3.9.yaml (hunk 2 above)
Comment (developer-guy) [reply postfix-3.9.yaml:42]: to be honest, I don't know how to test postfix package, this is where I'm searching help for, but I'll try for sure.

Comment (developer-guy) [reply postfix-3.9.yaml:52]: thanks to ChatGPT, I wrote some tests o7

File: postfix-3.9.yaml [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+  pipeline:
+    - runs: |
+        #!/bin/bash
+        #create postdrop group and postfix user with unique group IDs
+
+        addgroup -S postfix
+        addgroup -S postdrop
+        adduser -S -D -H -h /dev/null -s /sbin/nologin -G postfix postfix
+        chown -R postfix:postdrop /var/spool/postfix
+        chmod -R 755 /var/spool/postfix
```
Comment (imjasonh) [inline postfix-3.9.yaml:238]: This makes it seem like this user has to exist to run the package. Is that the case? If so, we might need to ensure the image that includes this package also has this user.

**Review by imjasonh — DISMISSED** [review]

File: postfix-3.9.yaml (hunk 3 above)
Comment (mamccorm) [reply postfix-3.9.yaml:238]: Good catching this in the testing btw. Given this is a package request and not an image request, it's be worth digging into this a little more, as for anyone who might want the package, they'd need to be aware

Comment (developer-guy) [reply postfix-3.9.yaml:238]: as we have this:

```
  scriptlets:
    pre-install: |
      #!/bin/sh
      addgroup -S postfix 2>/dev/null
      addgroup -S postdrop 2>/dev/null
      adduser -S -H -h /var/spool/postfix -G postfix -g postfix postfix 2>/dev/null
      addgroup postfix mail 2>/dev/null
      adduser -S -H -h /var/mail/domains -s /sbin/nologin -G postdrop -g vmail vmail 2>/dev/null
      exit 0
```

this shouldn't be any problem, am I right?

**Review by mamccorm — APPROVED** [review]

**Pull request merged**